crates/prism/src/webserver.rs

```diff
@@ -142,16 +146,33 @@ async fn get_hashchain(
     State(session): State<Arc<Sequencer>>,
     Json(request): Json<UserKeyRequest>,
 ) -> impl IntoResponse {
-    match session.get_hashchain(&request.id).await {
-        Ok(hashchain_or_proof) => match hashchain_or_proof {
-            Ok(hashchain) => (StatusCode::OK, Json(UserKeyResponse { hashchain })).into_response(),
-            Err(non_inclusion_proof) => {
-                (StatusCode::BAD_REQUEST, Json(non_inclusion_proof)).into_response()
-            }
-        },
-        Err(err) => (
+    let get_hashchain_result = session.get_hashchain(&request.id).await;
+    let Ok(hashchain_response) = get_hashchain_result else {
```

Comment:
this let..else is also cleaner because the variable assignment is clear and Ok doesn't feel as obfuscating

Reply:
Okay, I also feel like the length of this `HashchainResponse::` construct is making this difficult. Will bring that into scope.